classes/travel/TravelManager.php

```diff
@@ -276,10 +287,35 @@ public function fetchNearbyPlayers(): array {
                     level: 30,
                     battle_id: 0,
                     direction: $this->user->location->directionToTarget($placeholder_coords),
+                    distance: $this->user->location->distanceDifference($placeholder_coords),
                 );
             }
         }
 
+        usort($return_arr, function ($a, $b) {
+            if ($a->alignment == 'Enemy' && $b->alignment == 'Ally') {
+                return -1; // $a comes before $b
+            } elseif ($a->alignment == 'Ally' && $b->alignment == 'Enemy') {
+                return 1; // $b comes before $a
+            } else {
+                // Sort by distance first
+                if ($a->distance < $b->distance) {
+                    return -1; // $a comes before $b
+                } elseif ($a->distance > $b->distance) {
+                    return 1; // $b comes before $a
+                } else {
+                    // Sort by level if distances are equal
+                    if ($a->level > $b->level) {
+                        return -1; // $a comes before $b
+                    } elseif ($a->level < $b->level) {
+                        return 1; // $b comes before $a
+                    } else {
+                        return 0; // Objects are equal
+                    }
+                }
```

Comment:
Huh, I wonder if this could lead to more misclicks. The more factors we introduce that can cause players to move up and down on the list, the easier it is to misclick and attack the wrong player.

I'm okay trying this out if you don't feel that's a concern, but suspect it might be better to keep the order as consistent as possible

Reply:
I think the main issue (now that there's a visual indicator on the map) is the attack link itself moving, so I'm confident this will be felt as an improvement. I'll keep an eye on player feedback and revert/adjust if necessary.